vm/src/obj/objfunction.rs

```diff
@@ -379,6 +377,22 @@ impl PyBoundMethod {
     fn module(&self, vm: &VirtualMachine) -> Option<PyObjectRef> {
         vm.get_attribute(self.function.clone(), "__module__").ok()
     }
+
+    #[pymethod(magic)]
+    fn eq(&self, other: PyObjectRef) -> PyResult<PyComparisonValue> {
+        let other = class_or_notimplemented!(Self, other);
+        Ok(PyComparisonValue::Implemented(
+            self.function.is(&other.function) && self.object.is(&other.object),
+        ))
+    }
+
+    #[pymethod(magic)]
+    fn ne(&self, other: PyObjectRef) -> PyResult<PyComparisonValue> {
+        let other = class_or_notimplemented!(Self, other);
+        Ok(PyComparisonValue::Implemented(
+            !self.function.is(&other.function) || !self.object.is(&other.object),
+        ))
```

Comment:
when eq returns PyComparisonValue, we can add ne by using it
```suggestion
        Ok(self.eq(other)?.map(|v| !v))
```

it also looks like there is no path to return Err

Reply:
Oh, yeah, ther PyResult is there because `class_or_notimplemented!()` returns an `Ok(NotImplemented)` when it fails.